core/src/com/rallyhealth/weepickle/v1/core/Visitor.scala

```diff
@@ -102,6 +102,7 @@ trait Visitor[-T, +J] extends AutoCloseable {
     override def mapFunction(v: J): Z = f(v)
     def mapNonNullsFunction(v: J): Z = f(v)
   }
+  def mapKeys(f: CharSequence => CharSequence): Visitor[T, J] = new Visitor.MapKeys[T, J](Visitor.this, f)
```

Comment:
I'd rather keep this method definition outside of the minimal Visitor interface. Fine as an implicit extension method though. Should probably include "recursive" in the name. I could see use cases either way.

Reply:
I created VisitorImplicits to house typeclass for mapKeys. I did not include "recursive" in the name because Delegate and MapTo (accessed by "map" function) are also recursive and don't have "recursive" in their names, and wanted to stay consistent with that.